geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/utils/ThreadManagement.scala

```diff
@@ -12,63 +12,87 @@ import java.util.concurrent.{PriorityBlockingQueue, ScheduledThreadPoolExecutor,
 
 import com.google.common.util.concurrent.MoreExecutors
 import com.typesafe.scalalogging.LazyLogging
-import org.locationtech.geomesa.filter.filterToString
 import org.locationtech.geomesa.index.geotools.GeoMesaFeatureReader
+import org.locationtech.geomesa.utils.conf.GeoMesaSystemProperties.SystemProperty
+
+import scala.util.control.NonFatal
 
 /**
  * Singleton for registering and managing running queries.
  */
-object ThreadManagement extends LazyLogging {
+object ThreadManagement extends Runnable with LazyLogging {
 
-  private val interval = 5000L // how often we check for expired readers
-  private val ordering = new Ordering[ReaderAndTime]() {
-    // head of queue will be ones the will timeout first
-    override def compare(x: ReaderAndTime, y: ReaderAndTime) = x.killAt.compareTo(y.killAt)
-  }
-  private val openReaders = new PriorityBlockingQueue[ReaderAndTime](11, ordering) // size will grow unbounded
-
-  private val reaper = new Runnable() {
-    override def run() = {
-      val time = System.currentTimeMillis()
-      var loop = true
-      var numClosed = 0
-      while (loop) {
-        val holder = openReaders.poll()
-        if (holder == null) {
-          loop = false
-        } else if (holder.killAt < time) {
-          if (!holder.reader.isClosed) {
-            logger.warn(s"Stopping query on schema '${holder.reader.query.getTypeName}' with filter " +
-                s"'${filterToString(holder.reader.query.getFilter)}' based on timeout of ${holder.timeout}ms")
-            holder.reader.close()
-            numClosed += 1
-          }
-        } else {
-          if (!holder.reader.isClosed) {
-            openReaders.offer(holder)
+  // how often we check for expired readers
+  private val interval = SystemProperty("geomesa.query.timeout.check").toDuration.map(_.toMillis).getOrElse(5000L)
```

Comment:
Is this a new System property?  If so, docs?

Reply:
wilco